components/sidebar-navigation/Component.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from "react";
+import { cnb } from "cnbuilder";
+
+// import specific templates for the component
+import BurgerBar from "../../packages/SVG-library/BurgerBar";
+import Close from "../../packages/SVG-library/Close";
+import LinkItem from "./Components/LinkItem";
+
+/**
+ * Base component
+ *
+ * @param {string} title The component title
+ * ... any other options needed
+ * @returns {JSX.Element}
+ * @constructor
+ */
+
+export default function sidebarNavigation({
+  asset_assetid,
+  asset_short_name,
+  asset_url,
+  active,
+  menu,
+}) {
+  const [navOpen, setNavOpen] = useState(false);
+  const [showMobileButton, setShowMobileButton] = useState(false);
+
+  useEffect(() => {
+    if (window.innerWidth > 991) {
+      setShowMobileButton(false);
+      setNavOpen(true);
+    } else if (window.innerWidth < 991) {
+      setShowMobileButton(true);
+      setNavOpen(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    const handleResize = () => {
+      if (window.innerWidth > 991) {
+        setShowMobileButton(false);
+        setNavOpen(true);
+      } else if (window.innerWidth < 991) {
+        setShowMobileButton(true);
+        setNavOpen(false);
+      }
+    };
+
+    window.addEventListener("resize", handleResize);
+
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
+
+  return (
+    <>
+      {showMobileButton && (
+        <button
+          data-role="sidebar-navigation-toggle"
+          aria-controls="sidebar-navigation"
+          aria-label="sidebar-navigation"
+          aria-expanded={!!navOpen}
+          type="button"
+          onClick={() => setNavOpen(!navOpen)}
+          className={cnb(
+            "lg:su-hidden su-flex su-items-center su-w-full su-h-[5.6rem] su-p-10 su-text-left su-font-semibold su-border-2 su-border-digital-red su-text-digital-red dark:su-border-dark-mode-red dark:su-text-dark-mode-red",
+            navOpen &&
+              "su-bg-digital-red su-text-white dark:su-bg-dark-mode-red dark:aria-expanded:su-text-black-true"
+          )}
```

Comment:
The aria-expanded: prefixes work out nicely! ⭐ I suspect we might need hocus styles for the mobile button however, for both the close and open states.

Reply:
For a11y, it might be best if the menu collapses after the escape key is clicked. We have a custom `useEscape` hook for this purpose and have used it on many sites:
https://github.com/SU-SWS/ood-giving-campaign/blob/dev/hooks/useEscape.ts

To use the hook, you could do something like this:
https://github.com/SU-SWS/ood-giving-campaign/blob/5a155be90494b8f6ac3a906f7099ba1fec34e83f/components/ChangemakerCard/ChangemakerCard.tsx#L76

Perhaps check with SODA to see if this need to close on click outside as well. If you do, there's another hook useOnClickOutside from this library (which has a lot of useful hooks)
https://usehooks.com/

If you need just that particular one, we have one available here:
https://github.com/SU-SWS/saa_alumni/blob/dev/src/hooks/useOnClickOutside.js